### PR TITLE
Integrate transaction fee estimate endpoint

### LINKED ISSUE #4999: Integrate /transactions/estimate-fees endpoint
### Description

Integrate https://github.com/LiskHQ/lisk-service/issues/1573 and replace with existing fee calculation logic.

### Motivation

- Calculate transaction fee
- Calculate cross chain transfer fee
- Reduce code duplication on client side
- Accurate fee calculation

### PR DESCRIPTION
**Pull request by eniolam1000752**
### What was the problem?

This PR resolves #4999 #5110

### How was it solved?

- [x] Implement `useTransactionEstimateFees` hook
- [x] consume `useTransactionEstimateFees` into useTransactionFee hook
- [x] Separate form errors from transaction error displays
- [x] Implement dry run on continuing from a form step
- [x] Fix/update unit tests

### How was it tested?

- JEST
- Visually

### CODE REVIEW

File: src/modules/auth/components/EnterPasswordForm/decryptAccountWorker.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/* eslint-disable no-restricted-globals */
+import { decryptAccount } from '../../../account/utils';
```
Comment (ManuGowda) [inline src/modules/auth/components/EnterPasswordForm/decryptAccountWorker.js]: follow same file name convention `decryptAccount.worker.js`

File: src/modules/auth/components/EnterPasswordForm/decryptAccountWorker.test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { decryptAccount } from '../../../account/utils';
```
Comment (ManuGowda) [inline src/modules/auth/components/EnterPasswordForm/decryptAccountWorker.test.js]: same update the file name

File: src/modules/transaction/components/TxComposer/index.js [hunk 3]
```diff
@@ -57,23 +68,45 @@ const TxComposer = ({
...
-  const { minimumFee, components, transactionFee } = useTransactionFee({
+  const {
+    minimumFee,
+    components,
+    transactionFee,
+    messageFeeTokenID,
+    messageFee,
+    isFetched,
+    isLoading: isLoadingFee,
+  } = useTransactionFee({
```
Comment (ManuGowda) [inline src/modules/transaction/components/TxComposer/index.js:84]: Call fee hook once all the required form fields are available. Check delegate registration transaction, without entering the keys its triggering the hook

File: src/modules/transaction/hooks/useTransactionFee/useTransactionFee.js [hunk 4]
```diff
@@ -1,69 +1,61 @@
...
-  let txComponentType = '';
+  let components = [bytesFee, accountInitializationFee, priorityFee].filter(
+    (item) => item.value > 0
+  );
+
   if (
-    moduleCommand === MODULE_COMMANDS_NAME_MAP.transfer ||
-    moduleCommand === MODULE_COMMANDS_NAME_MAP.transferCrossChain
+    moduleCommand !== 'token:transfer' &&
+    moduleCommand !== 'token:transferCrossChain' &&
```
Comment (ManuGowda) [inline src/modules/transaction/hooks/useTransactionFee/useTransactionFee.js]: use the map MODULE_COMMANDS_NAME_MAP.transfer , MODULE_COMMANDS_NAME_MAP.transferCrossChain?

**Review by ManuGowda — COMMENTED** [review]
Show feedback on form which doesn't have enough balance to complete the transaction.

STR:
- have 0.1 balance on wallet
- initiate register multisignature/register validator/staking
- check the user has not received any feedback

File: src/modules/transaction/components/TxComposer/index.js [hunk 5]
```diff
@@ -8,13 +9,19 @@ import Box from 'src/theme/box';
...
 import { useTransactionFee } from '@transaction/hooks/useTransactionFee';
 import { PrimaryButton } from 'src/theme/buttons';
+import to from 'await-to-js';
+import { useCommandSchema } from 'src/modules/network/hooks';
+import useSettings from 'src/modules/settings/hooks/useSettings';
 import Feedback from './Feedback';
 import { getFeeStatus } from '../../utils/helpers';
-import { splitModuleAndCommand } from '../../utils';
+import { fromTransactionJSON, joinModuleAndCommand, splitModuleAndCommand } from '../../utils';
+import { dryRun } from '../../api';
```
Comment (oskarleonard) [inline src/modules/transaction/components/TxComposer/index.js]: can we use alias for these?

**Review by ManuGowda — CHANGES_REQUESTED** [review]
for fee estimate can we trigger the estimate endpoint once onBlur? so that we do not have api calls until the input is complete?
ex.  a user can type 64 char for message and we are calling the endpoint 64 times :disappointed: so its not optimised.. same applies for amount or any other field which needs blur implementation.

**Review by ManuGowda — APPROVED** [review]

**Review by oskarleonard — APPROVED** [review]

**Pull request merged**